src/rockstor/system/tests/test_osi.py

```diff
@@ -1665,6 +1665,7 @@ def test_scan_disks_lsblk_parse_fail(self):
                 'NAME="/dev/sdb3" MODEL="" SERIAL="" SIZE="2G" TRAN="" VENDOR="" HCTL="" TYPE="part" FSTYPE="swap" LABEL="SWAP" UUID="ee6de8d5-eb1c-4d19-943c-0d24ecee92a7"',  # noqa E501
                 'NAME="/dev/sdb4" MODEL="" SERIAL="" SIZE="29.9G" TRAN="" VENDOR="" HCTL="" TYPE="part" FSTYPE="btrfs" LABEL="ROOT" UUID="46befa9e-b399-480d-b10d-338a4c875615"',  # noqa E501
                 "",
+                'NAME="/dev/sdd" MODEL="SanDisk 3.2Gen1" SERIAL="01016c73a2fabf22051f78f30546627c9fb5e8083fbc14d477f85d6a269220dd15710000000000000000000060a25f5700804f0091558107a3ab6843" SIZE="57.3G" TRAN="usb" VENDOR=" USB    " HCTL="6:0:0:0" TYPE="disk" FSTYPE="" LABEL="" UUID=""',  # noqa E501
```

Comment:
@cellisten Just a note about this additional line. All good except that we purposefully try to mimic the output of our command rapper: which notably has an empty line as it's last line of output. Hence the "" entry there. So your additional line should belong before, not after that "" entry. Also note that the order of output can be important. Hence us predominantly using real-world output where applicable. But I don't think that is important in this case as there is no partition info for this extra drive: these normally come after the parent block device (whole device).

Thanks for continuing to chip away at this one. Again much appreciated.

Have you run all of the tests with your changes in-play? That may be informative.

Also, what long serial numbers some of those SanDisk drive have. And of course this new quirk re spaces all over the place around  in VENDOR value (up-front even now!). Sometimes I think they just make this stuff up :).

Reply:
Honestly, lsblk might do well to sanitize some of the more non-machine readable output :)
I have run the full barrage of tests without any issues with the change. I will move the entry. I was wondering about that final empty one and first thought it was a test case in of itself but now I understand it.

Reply:
Agreed, some of these commands have a json output option. But we are also still releasing rpms for older Leap versions. So that can mean that the newer commands that have these more modern option modes are not yet available.

But all in good time hopefully. Plus there are ongoing improvements in some Python libraries native to btrfs that should help lift us out of some of this more messy human readable command output fragility. But in a way I also like that we run 'real world' commands that folks can also run themselves to test output outside of our parsing. Swings and roundabouts.